docs/operations/clean-metadata-store.md

```diff
@@ -0,0 +1,168 @@
+
+Datasource cleanup uses the following configuration:
+ - `druid.coordinator.kill.datasource.on`: When `true`, enables cleanup datasources created by supervisors.
+ - `druid.coordinator.kill.datasource.period`: Defines the frequency in [ISO 8601 format](https://en.wikipedia.org/wiki/ISO_8601#Durations) for the cleanup job to check for and delete eligible datasource records. Defaults to `P1D`.
+ - `druid.coordinator.kill.datasource.durationToRetain`: Defines the retention period in [ISO 8601 format](https://en.wikipedia.org/wiki/ISO_8601#Durations) after creation that datasource records become eligible for deletion.
+
+### Indexer task logs
+The Druid Overlord handles task log metadata management.
+
+Indexer task log cleanup on the Overlord uses the following configuration:
+- `druid.indexer.logs.kill.enabled`: When `true`, enables cleanup of task logs.
+- `druid.indexer.logs.kill.durationToRetain`: Defines the length of time in milliseconds to retain task logs.
+- `druid.indexer.logs.kill.initialDelay`: Defines the length of time in milliseconds after the Overlord starts before it executes its first job to kill task logs.
+- `druid.indexer.logs.kill.delay`: The length of time in milliseconds between jobs to kill task logs.
+
+For more detail, see [Task logging](../configuration/index.md#task-logging).
+
+## Automated metadata cleanup example configuration
+Consider a scenario where you have scripts to create and delete hundreds of datasources and related entities a day. You do not want to fill your metadata store with leftover records. The datasources and related entities tend to persist for only one or two days. Therefore, you want to run a cleanup job that identifies and removes leftover records that are at least four days old. The exception is for audit logs, which you need to retain for 30 days:
+
+```
+...
+# Schedule the metadata management store task for every hour:
+druid.coordinator.period.metadataStoreManagementPeriod=P1H
+
+# Set a kill task to poll every day to delete Segment records
+# and segments in deep storage > 4 days old.
+# Required also for automated cleanup of rules
+# and compaction configuration.
+
+druid.coordinator.kill.on=true
+druid.coordinator.kill.period=P1D 
+druid.coordinator.kill.durationToRetain=P4D
+druid.coordinator.kill.maxSegments=1000
+
+# Poll every day to delete audit records > 30 days old
+druid.coordinator.kill.audit.on=true
+druid.coordinator.kill.audit.period=P1D
+druid.coordinator.kill.audit.durationToRetain=P30D
+
+# Poll every day to delete supervisor records > 4 days old
+druid.coordinator.kill.supervisor.on=true
+druid.coordinator.kill.supervisor.period=P1D
+druid.coordinator.kill.supervisor.durationToRetain=P4D
+
+# Poll every day to delete rules records > 4 days old
+druid.coordinator.kill.rule.on=true
+druid.coordinator.kill.rule.period=P1D
+druid.coordinator.kill.rule.durationToRetain=P4D
+
+# Poll every day to delete compaction configuration records > 4 days old
```

Comment:
compaction configuration auto cleanup does not have durationToRetained. You cannot set > 4 days. If you turn it on, then compaction config is eligible for cleanup right away when the associated datasource has no segment (segment killed by kill task)

Reply:
This should just say `# Poll every day to delete compaction configuration records for datasource without segments` or just `# Poll every day to delete compaction configuration records`